software/contrib/strange_attractor.py

```diff
@@ -0,0 +1,384 @@
+from europi import *
+from time import ticks_diff, ticks_ms
+from math import fabs, floor
+from random import choice
+
+'''
+Strange Attractor
+author: Sean Bechhofer (github.com/seanbechhofer)
+date: 2022-03-15
+labels: gates, triggers, randomness
+
+Strange Attractor is a source of chaotic modulation. It can use a variety of different implementations.
+
+Outputs 1, 2 and 3 are based on the x, y and z values generated by the attractor.
+
+Outputs 4, 5 and 6 are gates based on the values of x, y and z and relationships between them. 
+
+digital_in: Pause motion when HIGH
+analog_in: 
+
+knob_1: Adjust speed
+knob_2: Adjust threshold for triggers
+
+button_1: Decrease output voltage range, change equation system
+button_2: Increase output voltage range
+
+output_1: x 
+output_2: y
+output_3: z
+output_4: triggers/gates
+output_5: triggers/gates
+output_6: triggers/gates
+
+'''
+# Version number
+
+VERSION="1.0"
+
+# Overclock the Pico for improved performance.
+machine.freq(250_000_000)
+
+# Maximum voltage output. Cranking this up may cause issues with some modules. 
+MAX_OUTPUT = 5
```

Comment:
I'm curious why you made the range so easy to change with the buttons, but capped it at half of the eurpoi's output range? 

Using `europi.MAX_OUTPUT_VOLTAGE` instead of this constant will give users flexibility, and the buttons (or an external attenuator) will allow them to configure it to a lower range if they are using a module that the comment above is referring to.

This isn't anythinf to hold up a merge for, just something I noticed after actually running the script.

Reply:
It was rather the other way round -- I capped it, then added facility to change. When I was first prototyping the app, I was getting a lot of noise/clipping when sending CV to some modules. I experimented with dialling down the range and this seemed to fix it. Then when I was fleshing out the UI I had the idea of using the buttons to change the range and didn't think to go back and extend the MAX_OUTPUT. 

I'll make the change you suggest when tweaking for menu compatibility. Range value could then be something to store in the saved settings.